src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/controllers/SubjectAccessRequestController.kt

```diff
@@ -0,0 +1,108 @@
+
+  @GetMapping
+  @Operation(summary = "Provides content for a prisoner to satisfy the needs of a subject access request on their behalf")
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Request successfully processed - content found",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = SubjectAccessRequestContent::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "204",
+        description = "Request successfully processed - no content found",
+      ),
+      ApiResponse(
+        responseCode = "209",
+        description = "Subject Identifier is not recognised by this service",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "The client does not have authorisation to make this request",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "500",
+        description = "Unexpected error occurred",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  fun getSarContentByReference(
+    @RequestParam(name = "prn", required = false)
+    @Parameter(description = "NOMIS Prison Reference Number")
+    prn: String?,
+    @RequestParam(name = "crn", required = false)
+    @Parameter(description = "nDelius Case Reference Number")
+    crn: String?,
+    @RequestParam(value = "fromDate", required = false)
+    @Parameter(description = "Optional parameter denoting minimum date of event occurrence which should be returned in the response")
+    fromDate: LocalDate?,
+    @RequestParam(value = "toDate", required = false)
+    @Parameter(description = "Optional parameter denoting maximum date of event occurrence which should be returned in the response")
+    toDate: LocalDate?,
+  ): ResponseEntity<Any> {
+    if (!prn.isNullOrBlank()) {
+      return service.getContentFor(prn, fromDate, toDate)
+        ?.let { ResponseEntity.ok(it) }
+        ?: ResponseEntity.status(HttpStatus.NO_CONTENT)
+          .body(
+            ErrorResponse(
+              status = HttpStatus.NO_CONTENT,
+              userMessage = "Request successfully processed - no content found",
+              developerMessage = "Request successfully processed - no content found",
+            ),
+          )
+    }
+    return ResponseEntity.status(209).body(
```

Comment:
not really clear what should happen if both `prn` and `crn` supplied, but I've gone for a `209` if `prn` is missing 🤷

Reply:
Feels like a 4xx but response codes seem odd for SAR anyway